app/src/androidTest/java/com/github/sdp/mediato/AuthenticationActivityTest.java

```diff
@@ -135,11 +149,46 @@ public void testLogInButtonWorks() throws InterruptedException {
         }
 
         Thread.sleep(3000);
```

Comment:
Why do we need the Thread.sleep here exactly ? Doesn't login provide us with a future or something like that ?

Reply:
We actually don't, so I removed all of them and used futures when needed